### DISCUSSION TITLE
Is it possible to propagate comments to autogenerated methods with `@DependencyClient`?

### DISCUSSION THREAD

**Discussion by rudedogdhc** [pointfreeco/swift-dependencies#194] [Ideas]
Before `DependencyMacros` I would often design a dependency thus:

```
public struct MyDependency {
  var frobAll: (_ count: Int) -> Void

  /// Does the frobbing `count` number of times.
  /// - Parameter count: The number of times to frob.
  ///
  /// Further discussion of what this is going to do.
  public func frobAll(count: Int) {
    frobAll(count)
  }
}
```

With the addition of `DependencyClient`, the convenience method can be generated for me, so it's much easier to get mostly the same result simply with

```
@DependencyClient
public struct MyDependency {
  public var frobAll: (_ count: Int) -> Void
}
```

The bad news is that I no longer have the documentation comment for the method. I could of course attach the documentation comment to the closure property, but that only helps callers who call the closure, not the people who call the auto-generated method.

I'm not even sure that this can be done, but is it possible to propagate the comments on the properties to these auto-generated methods? And even if it was possible, would Xcode even notice documentation comments inside macro expansions?

**Comment by fruitcoder** [comment]
The biggest issue here is that functions added by macros are not surfaced in the documentation (as of Xcode 16.3 RC2). So your second code snippet just looks like this in the documentation window:

<img width="637" alt="image" src="https://github.com/user-attachments/assets/0870a8ac-dd69-4891-8611-3086f5fa8190" />

Once Xcode provides this we could just copy the `docBlockComment`s and `docLineComment`s to the `@DependencyEndpoint` functions.

**Reply by fruitcoder** [reply]
unfortunately this still hasn't changed in Xcode 26 Beta 4 😢

**Reply by mbrandonw** [reply]
Hi @fruitcoder, have you filed an issue on DocC about this?

**Reply by fruitcoder** [reply]
I could have sworn I did but can't find it right now. But now I filed it now (again) FB19381490
